nilearn/regions/hierarchical_kmeans_clustering.py

```diff
@@ -337,6 +352,15 @@ def transform(
         """
         check_is_fitted(self)
 
+        if not SKLEARN_LT_1_6:
```

Comment:
May the check can be done as:

```
if SKLEARN_LT_1_6:
```

in the same order as the previous function.